packages/openapi2-parser/lib/schema.js

```diff
@@ -114,16 +114,29 @@ class DataStructureGenerator {
     // Create member elements for required keys which are not defined in properties
     const missingRequiredProperties = required.filter(name => properties[name] === undefined);
     const requiredMembers = missingRequiredProperties.map((name) => {
-      const member = new this.minim.elements.Member(name);
+      let member;
+
+      if (schema.additionalProperties !== undefined && typeof schema.additionalProperties !== 'boolean') {
+        member = this.generateMember(name, schema.additionalProperties);
+      } else {
+        member = new this.minim.elements.Member(name);
+      }
+
       member.attributes.set('typeAttributes', ['required']);
       return member;
     });
     element.content.push(...requiredMembers);
 
-    if (schema.additionalProperties === false) {
+    if (schema.additionalProperties !== undefined && schema.additionalProperties !== true) {
       const typeAttributes = element.attributes.get('typeAttributes') || new this.minim.elements.Array([]);
-      typeAttributes.push('fixed-type');
+      typeAttributes.push('fixedType');
```

Comment:
I mistakenly got the case incorrect, `fixed-type` is the human readable name in MSON, `fixedType` is the casing for API Elements.